test/Tests/src/Semantic/Case_Spec.enso

```diff
@@ -47,6 +47,19 @@ spec = Test.group "Pattern Matches" <|
         case Float of
             Float -> Nothing
             _ -> Test.fail "Expected the Float constructor to match."
+    Test.specify "match on the Number literal" <|
+        foo x = case x of
+            1 -> "ONE"
+            2 -> "TWO"
+            200 -> "TWO HUNDRED"
+            9999 -> "NINE THOUSAND NINE HUNDRED NINETY NINE"
+            _ -> Test.fail "OTHER - "+x.to_text
```

Comment:
This test does not cover the BigInteger nor Double branches, I think we should add test cases for these two as well.

Reply:
Done in 7541e62

Reply:
Perfect, thank you